mava/networks.py

```diff
@@ -93,22 +96,60 @@ def __call__(self, obs_embedding: chex.Array, observation: Observation) -> distr
                 `step_count`.
 
         Returns:
-            A distrax.Categorical distribution over the action space for sampling actions from.
+            A transformed tfd.categorical distribution on the action space for action sampling.
 
         NOTE: We pass both the observation embedding and the observation object to the action head
         since the observation object contains the action mask and other potentially useful
         information.
         """
 
-        actor_logits = nn.Dense(self.action_dim, kernel_init=orthogonal(0.01))(obs_embedding)
+        actor_logits = nn.Dense(self.env.action_dim, kernel_init=orthogonal(0.01))(obs_embedding)
 
         masked_logits = jnp.where(
             observation.action_mask,
             actor_logits,
             jnp.finfo(jnp.float32).min,
         )
 
-        return distrax.Categorical(logits=masked_logits)
+        #  We transform this distribution with the `Identity()` transformation to
+        # keep the API identical to the ContinuousActionHead.
+        return IdentityTransformation(distribution=tfd.Categorical(logits=masked_logits))
+
+
+class ContinuousActionHead(nn.Module):
+    """ContinuousActionHead using a transformed Normal distribution."""
+
+    env: Environment
```

Comment:
In general, I think we shouldn't be passing envs in to network heads. We should rather just get the information we need beforehand and then pass that in.

Reply:
Maybe rather action_spec then, we should be able to get everything we need from this for both continuous and discrete

Reply:
Yeah then we make default values for both networks and mention in the discrete one that this is needed only for continuous case